src/dotnet-install.sh

```diff
@@ -618,41 +618,89 @@ get_specific_product_version() {
     if [ $# -gt 2  ]; then
         local package_download_link="$3"
     fi
-    local specific_product_version=$specific_version
+    local specific_product_version=null
+
+    local download_links=($(get_specific_product_version_url "$azure_feed" "$specific_version" true "$package_download_link")
+        $(get_specific_product_version_url "$azure_feed" "$specific_version" false "$package_download_link"))
+
+    for download_link in "${download_links[@]}"
+    do
+        say_verbose "Checking for the existence of $download_link"
+
+        if machine_has "curl"
+        then
+            specific_product_version=$(curl -s --fail "$download_link")
+            if [ $? -ne 0 ]; then
+                specific_product_version=null
+            else
+                break
```

Comment:
I would consider returning here instead. It would simplify logic, no need to set null.